test/e2e/vsphere/machines.go

```diff
@@ -0,0 +1,288 @@
+package vsphere
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/api/machine/v1beta1"
+	machinesetclient "github.com/openshift/client-go/machine/clientset/versioned/typed/machine/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/openshift/machine-api-operator/pkg/controller/vsphere"
+	util "github.com/openshift/machine-api-operator/test/e2e"
+)
+
+const (
+	machineRole         = "e2e-test"
+	machineReadyTimeout = time.Minute * 6
+)
+
+type MachineTest struct {
+	testCase    string
+	machineName string
+	dataDisks   []v1beta1.VSphereDisk
+	ctx         context.Context
+	cfg         *rest.Config
+	c           *kubernetes.Clientset
+	dc          *dynamic.DynamicClient
+	mc          *machinesetclient.MachineV1beta1Client
+}
+
+type MachineSetTest struct {
+	testCase  string
+	msName    string
+	dataDisks []v1beta1.VSphereDisk
+	ctx       context.Context
+	cfg       *rest.Config
+	c         *kubernetes.Clientset
+	dc        *dynamic.DynamicClient
+	mc        *machinesetclient.MachineV1beta1Client
```

Comment:
A lot of this doesn't actually vary, so does it need to be set provided directly in the test struct?

Reply:
I was using similar to our operator contexts that pass it around instead of having so many parameters in our helper functions.  I can look into trimming this out if you think its needed.

Reply:
Well, they are all global variables already so I wasn't sure if they needed to be passed in even, vs sourcing them from the global scope within the test function

Reply:
I ended up removing these.  I think I streamlined it much more now.